db/users/update.go

```diff
@@ -36,18 +36,27 @@ func UpdateUser(ctx context.Context, app, addonUUID, username string) error {
 		return errors.Wrap(ctx, err, "list the database's users")
 	}
 
-	// Check if the user exists
+	// Check if the user exists and is not protected
 	var userExists bool
+	var userProtected bool
 	for _, user := range l {
 		if user.Name == username {
 			userExists = true
+
+			if user.Protected {
+				userProtected = true
+			}
 			break
 		}
 	}
 	if !userExists {
 		return errors.New(ctx, fmt.Sprintf("User \"%s\" does not exist", username))
 	}
 
+	if userProtected {
+		return errors.New(ctx, fmt.Sprintf("User \"%s\" is protected and cannot be updated", username))
+	}
+
```

Comment:
For better readability I suggest:
```suggestion
    var user
    for _, u := range l {
	    if u.Name == username {
		    user = u
		    break
	    }
    }
    if user == nil {
	    return errors.New(ctx, fmt.Sprintf("User \"%s\" does not exist", username))
    }
    if user.Protected {
	    return errors.New(ctx, fmt.Sprintf("User \"%s\" is protected", username))
    }
```
WDYT?

Reply:
I pushed a suggestion, WDYT?